### ISSUE TITLE
Content Script injection fails on 51.0.1

### ISSUE THREAD

**Issue by ranjanbipul** [mdn/webextensions-examples#185]
When this addon is installed after signing from any website, it doesn't work on mozilla firefox 51.0.1. It gives following error when injecting content script.
```

Error opening input stream (invalid filename?): moz-extension://624fbb28-a2ef-49ad-8081-6632ac2578e6/borderify.js  ExtensionContent.jsm:226
uncaught exception: Error opening input stream (invalid filename?): moz-extension://624fbb28-a2ef-49ad-8081-6632ac2578e6/borderify.js
```

I was facing same issue with my own addon and then tried this simple addon and it didn't work.

**Comment by wbamberg** [comment]
I can install the build, signed version from here: https://github.com/mdn/webextensions-examples/blob/master/build/borderify-1.0-an%2Bfx.xpi and it works just fine (Firefox 51.0.1, Mac OS X).

Can you please check whether you can reproduce with this version?

Can you please share the XPI that doesn't install, so we could take a look at it?

Is it only signed XPIs that fail, or do unsigned ones fail too? Does it work when you use ["Load Temporary Add-on"](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Temporary_Installation_in_Firefox)?

**Comment by ranjanbipul** [comment]
The issue got fixed in Firefox 52.0.

When "Loading temporary add-on" it was working fine. But, the signed one was having issue.
